src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,14 +1,38 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { actions as filterActions } from '../../features/filter';
+import { Status } from '../../types/Status';
 
 export const TodoFilter: React.FC = () => {
+  const dispatch = useDispatch();
+  const [searchQuery, setSearchQuery] = useState('');
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setSearchQuery(event.target.value);
+    dispatch(filterActions.addQuery(event.target.value));
+  };
+
+  const handleInputClear = () => {
+    setSearchQuery('');
+    dispatch(filterActions.clearQuery());
+  };
+
+  const hanleStatusChange = (value: Status) => {
+    dispatch(filterActions.addStatus(value));
+  };
+
   return (
-    <form
-      className="field has-addons"
-      onSubmit={event => event.preventDefault()}
-    >
+    <form className="field has-addons">
       <p className="control">
         <span className="select">
-          <select data-cy="statusSelect">
+          <select
+            data-cy="statusSelect"
+            onChange={(e) => {
```

Comment:
Avoid making your jsx complicated, consider using a separate event handler
```suggestion
            onChange={onSelect}
```